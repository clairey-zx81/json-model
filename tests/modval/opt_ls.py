# Generated for model: ./opt_ls.model.json
#
# Generated by jmc (JSON Model Compiler) (https://github.com/clairey-zx81/json-model)
#

from typing import Callable
import re as re
import datetime
import urllib.parse

type Jsonable = None|bool|int|float|str|list[Jsonable]|dict[str, Jsonable]
type CheckFun = Callable[[Jsonable, str], bool]
type PropMap = dict[str, CheckFun]
type TagMap = dict[None|bool|float|int|str, CheckFun]

json_model_0_must: PropMap

# define "json_model_0_must_i" ($.i)
def jm_f_0(value: Jsonable, path: str) -> bool:
    # $.i
    result = ((isinstance(value, int) and not isinstance(value, bool)) or (isinstance(value, float) and int(value) == value))
    return result

# define "json_model_0_must_f" ($.f)
def jm_f_1(value: Jsonable, path: str) -> bool:
    # $.f
    result = isinstance(value, float)
    return result



# object $
def json_model_0(value: Jsonable, path: str) -> bool:
    if not isinstance(value, dict):
        return False
    must_count = 0
    for prop, model in value.items():
        assert isinstance(prop, str)
        if prop in json_model_0_must:  # must
            must_count += 1
            if not json_model_0_must[prop](model, f"{path}.{prop}"):
                return False
        else:  # no catch all
            return False
    return must_count == 2



# entry function check_model
def check_model(value: Jsonable, path: str = "$") -> bool:
    return json_model_0(value, path)


# object properties maps
json_model_0_must = {
    "i": jm_f_0,
    "f": jm_f_1,
}


