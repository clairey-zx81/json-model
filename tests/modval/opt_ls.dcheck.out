opt_ls.0.true.json: PASS
opt_ls.1.true.json: PASS
opt_ls.2.false.json: FAIL [('.f', '$.f', 'expecting a number')]
opt_ls.3.false.json: FAIL [('.f', '$.f', 'expecting a number')]
opt_ls.4.false.json: FAIL [('.i', '$.i', 'expecting a (loose) integer')]
